### DISCUSSION TITLE
How to get last messages from a channel in using discord.py

### DISCUSSION THREAD

**Discussion by DakshYadav358** [Rapptz/discord.py#7019] [Q&A]
How to get the last messages from a channel in using discord.py

**Comment by Rapptz** [comment]
You can get them with either [`TextChannel.last_message`](https://discordpy.readthedocs.io/en/latest/api.html#discord.TextChannel.last_message) or [`TextChannel.last_message_id`](https://discordpy.readthedocs.io/en/latest/api.html#discord.TextChannel.last_message_id).

**Reply by DakshYadav358** [reply]
what is TextChannel in this is it a predefined thing or something the user defines??

**Reply by Rapptz** [reply]
`TextChannel` here would be an object that is an instance of the type `TextChannel`. Something like `message.channel` or `ctx.channel` or whatever. Though these can be different types than `TextChannel`.

**Comment by adhxtry** [comment]
The`last_message` attribute isn't always correct, use `|coro|` `TextChannel.history()` for fetching the last message accurately.
But yeah u can use last_message too, but for some lazy fetching or if you have message cahce off use history.
I recommend `last_message` if accuracy isnt a priority. `history()` is slow.

**Reply by lgaan** [reply]
It’s never not correct, it’s just optional since it relies on the cache.

It’s best to not jump straight to `fetch_message`, instead use `last_message_id` and fallback to fetch if it returns None.

This discussion is also already answered above, there’s no need to repeat already given information.
